### DISCUSSION TITLE
Using exceptions formatting in serilog logging library

### DISCUSSION THREAD

**Discussion by kolpav** [spectreconsole/spectre.console#817] [Q&A] [answered]
I really like how spectre is able to format exceptions especially its ExceptionFormats.ShortenEverything option which makes stack traces much more readable so I was wondering if anyone was able to somehow take advantage of this feature in logging libraries.

After quick googling I came to conclusion its not possible? There is no place in serilog where you can easily format exceptions and even if there was there is no way how to make spectre console to write to anything else but console right?

Having ability to get rid of useless paths in my stack traces is something I wanted to do for some time so if there are other options please share them.

**Comment by patriksvensson** [comment, marked as answer]
I think you will need to implement your own Serilog log sink for this, but I'm no Serilog expert.

**Reply by phil-scott-78** [reply]
If you were wanting to edit the stack trace regardless of sink I think an enricher would work. Couldn't reuse anything in Spectre, but it would be a starting point if you wanted to roll your own.
